examples/mini-app/src/components/game/DiceGame.tsx

```diff
@@ -0,0 +1,102 @@
+import diceBackground from "../../assets/game/game-background.png"
+
+import { CASINO_GAME_TYPE, DiceNumber, FORMAT_TYPE, formatRawAmount } from "@betswirl/sdk-core"
+import { useGameLogic } from "../../hooks/useGameLogic"
+import { DiceGameControls } from "./DiceGameControls"
+import { GameFrame } from "./GameFrame"
+import { GameConnectWallet } from "./shared/GameConnectWallet"
+import { BaseGameProps } from "./shared/types"
+
+export interface DiceGameProps extends BaseGameProps {}
+
+export function DiceGame({
+  theme = "system",
+  customTheme,
+  backgroundImage = diceBackground,
+  ...props
+}: DiceGameProps) {
+  const gameLogic = useGameLogic({
+    gameType: CASINO_GAME_TYPE.DICE,
+    defaultSelection: 50 as DiceNumber,
+    backgroundImage,
+  })
+
+  const {
+    isWalletConnected,
+    balance,
+    tokenDecimals,
+    areChainsSynced,
+    gameHistory,
+    refreshHistory,
+    betAmount,
+    selection: selectedNumber,
+    setSelection: setSelectedNumber,
+    betStatus,
+    gameResult,
+    formattedVrfFees,
+    gasPrice,
+    targetPayoutAmount,
+    multiplier,
+    isInGameResultState,
```

Comment:
Consider adding validation for the dice number bounds (1-99) when setting the selection:
\\\